Mathlib/NumberTheory/NumberField/Basic.lean

```diff
@@ -339,8 +339,8 @@ theorem ker_algebraMap_eq_bot : RingHom.ker (algebraMap (𝓞 K) (𝓞 L)) = ⊥
 theorem algebraMap.injective : Function.Injective (algebraMap (𝓞 K) (𝓞 L)) :=
   (RingHom.injective_iff_ker_eq_bot (algebraMap (𝓞 K) (𝓞 L))).mpr (ker_algebraMap_eq_bot K L)
 
-instance : NoZeroSMulDivisors (𝓞 K) (𝓞 L) :=
-  NoZeroSMulDivisors.of_algebraMap_injective (algebraMap.injective K L)
+instance : FaithfulSMul (𝓞 K) (𝓞 L) :=
+  (faithfulSMul_iff_algebraMap_injective (𝓞 K) (𝓞 L)).mpr <| algebraMap.injective K L
```

Comment:
Would you like to reinstate this as well?

Reply:
Done